src/submodules/io/csv_reader.jl

```diff
@@ -30,3 +30,12 @@ function populate_from_csv!(
         end
     end
 end
+
+function dump_to_csv(filename, agents::Vector{A}, fields = collect(fieldnames(A)); kwargs...) where {A<:AbstractAgent}
```

Comment:
the restriction to `Vector` is not good. The most common scenario would be to call this function on `agents(model)` which returns a value iterator.

I have a suggestion/advice: change your approach to writing functions. Start development on an interactive environment in VS Code where you run every single line and see what it does. Makes you understand and design code much much better.

Reply:
After you have the algorithm doing what you want, put it in a function body.

Reply:
I'll keep that in mind now on